crates/ruff_linter/src/rules/flake8_comprehensions/fixes.rs

```diff
@@ -937,7 +937,10 @@ pub(crate) fn fix_unnecessary_comprehension_in_call(
     let whitespace_after_arg = match &call.args[0].comma {
         Some(comma) => {
             let whitespace_after_comma = comma.whitespace_after.clone();
-            call.args[0].comma = None;
+            call.args[0].comma = Some(Comma {
+                whitespace_after: ParenthesizableWhitespace::default(),
+                ..comma.clone()
+            });
```

Comment:
Without this, we were (1) dropping commas in some tests (see fixture changes), and (2) adding an extra space after the comma when fixing `sum([x.val for x in bar], 0)` (not sure why?).